user-interface/src/main/java/life/qbic/datamanager/views/notifications/toasts/Toast.java

```diff
@@ -0,0 +1,259 @@
+        }
+      };
+      UI ui = attachEvent.getUI();
+      Registration beforeLeaveListenerRegistration = ui.addBeforeLeaveListener(beforeLeaveListener);
+      closeOnNavigationListeners.add(beforeLeaveListenerRegistration);
+    };
+  }
+
+  public void setDuration(Duration duration) {
+    super.setDuration((int) duration.toMillis());
+  }
+
+  /**
+   * The toast contains a success message
+   *
+   * @return
+   */
+  public Toast success() {
+    setType(Level.SUCCESS);
+    return this;
+  }
+
+  /**
+   * The toast contains an informational message only
+   *
+   * @return
+   */
+  public Toast info() {
+    setType(Level.INFO);
+    return this;
+  }
+
+  @Override
+  public void open() {
+    var cssClass = switch (level) {
+      case SUCCESS -> "success-toast";
+      case INFO -> "info-toast";
+    };
+    addClassName(cssClass);
+    super.open();
+  }
+
+  private void setType(Level level) {
+    this.level = requireNonNull(level, "type must not be null");
+  }
+
+
+  /**
+   * Creates a toast showing the component.
+   *
+   * @param content
+   * @return
+   */
+  static Toast create(Component content) {
+    requireNonNull(content, "content must not be null");
+    Toast toast = new Toast();
+    toast.withContent(content);
+    return toast;
+  }
+
+  /**
+   * Expands the toast and includes a link to a specific route target.
+   *
+   * @param linkText         The text of the link shown to the user
+   * @param navigationTarget the target of the navigation
+   * @param routeParameters  the parameters used for navigation
+   * @return
+   */
+  public Toast withRouting(String linkText, Class<? extends Component> navigationTarget,
+      RouteParameters routeParameters) {
+    return this.withContent(Toast.createRoutingContent(this.content, createRoutingComponent(
+        linkText, navigationTarget, routeParameters)));
+  }
+
+  /**
+   * Sets the content of the toast
+   *
+   * @param content the content of the toast to set.
+   * @return
+   */
+  private Toast withContent(Component content) {
+    requireNonNull(content, "content must not be null");
+    //we only want to remove the current content from its parent if the parent is not the new content.
+    //Otherwise the current content would be removed from the new content as well.
+    if (nonNull(this.content) && !ComponentFunctions.isParentOf(content, this.content)) {
+      this.content.removeFromParent();
+    }
+    this.content = content;
+    this.content.addClassName("toast-content");
+    refresh();
+    return this;
+  }
+
+  private void refresh() {
+    removeAll();
+    add(this.content, this.closeButton);
+  }
+
+  /**
+   * Can this toast be closed by the user?
+   *
+   * @return
+   */
+  public boolean isCloseable() {
+    return closeable;
+  }
+
+  /**
+   * Specifies whether the toast includes a button for the user to close the toast.
+   *
+   * @param closeable
+   * @return
+   */
+  public Toast setCloseable(boolean closeable) {
```

Comment:
What are the use cases for making that mutable? Should we not always offer the possibility for a user to close the toast?

Reply:
Wasn't clear at the time. I aggree